src/components/Mapper/Mapper.jsx

```diff
@@ -0,0 +1,753 @@
+  const [pokemonName, setPokemonName] = useState('');
+  const completedPokemonName = useDebouncedValue(pokemonName, 1500);
+
+  let originalImageData = {
+    highlight: {},
+    select: {},
+    enc: {},
+  };
+  let previousRectangle = {
+    highlight: null,
+    select: null,
+    enc: null,
+  };
+
+  let colorSettings = {
+    hov: { r: 247, g: 100, b: 200, a: 0.7 },
+    sel: { r: 72, g: 113, b: 247, a: 0.8 },
+    enc: { r: 247, g: 0, b: 0, a: 0.7 },
+  }
+
+  const [locationList, setLocationList] = useState([]);
+  const [showSettings, setShowSettings] = useState(false);
+  const [colors, setColors] = useState({
+    hov: { r: 247, g: 148, b: 72, a: 0.7 },
+    sel: { r: 72, g: 113, b: 247, a: 0.8 },
+    enc: { r: 247, g: 0, b: 0, a: 0.7 },
+  });
+
+  const [encounterList, setEncounterList] = useState({GroundEnc: [], SurfEnc: [], RodEnc: []});
+  const [trainerList, setTrainerList] = useState([]);
+  const [fieldItemsList, setFieldItems] = useState([]);
+  const [hiddenItemsList, setHiddenItems] = useState([]);
+  const [shopItemsList, setShopItems] = useState([]);
+  const [scriptItemsList, setScriptItems] = useState([]);
+  const [fixedShopList, setFixedShops] = useState([]);
+  const [heartScaleShopList, setHeartScaleShop] = useState([]);
+
+  const canvasRef = useRef(null);
+  const CLEAR_MODE = {
+    HIGHLIGHT: "highlight",
+    SELECT: "select",
+    ENCOUNTER: "enc",
+  }
+  //Component onMount
+  useEffect(() => {
+    console.log('Mounting...')
+    const context = canvasRef.current.getContext('2d', {willReadFrequently: true});
+    const image = new Image();
+    image.src = require('@site/static/img/new_small_mapper.png').default;
+    image.onload = () => {
+      context.drawImage(image, 0, 0);
+      drawOverlay(context);
+    };
+    console.log('Finish Mounting...', rect, canvasRef)
+
+  }, []) // Empty dependency array means this effect runs once after the initial render
+
+  function drawOverlay(ctx) {
+    coordinates.forEach(coord => {
+      // Draw zone outlines
+      ctx.beginPath();
+      ctx.moveTo(coord.x, coord.y);
+      ctx.lineTo(coord.x + coord.width, coord.y);
+      ctx.lineTo(coord.x + coord.width, coord.y + coord.height);
+      ctx.lineTo(coord.x, coord.y + coord.height);
+      ctx.closePath();
+      ctx.strokeStyle = 'rgba(0, 0, 0, 1)';
+      ctx.lineWidth = 1;
+      ctx.stroke();
+    });
+  };
+
+  /** Listener Events for the Canvas updates */
+  const updateLocationDataFromDropdown = (event) => {
+    // This is using the custom event that was created by the SearchBar
+    // After adding a listener, this is used to directly interact with the canvas
+    // This CANNOT pull the state from a useState or other async func.
+    // You must add a customEvent alongside where a state is updated
+    const selectedName = event.detail;
+    if (!selectedName) {
+      return;
+    }
+    const location = getLocationCoordsFromName(selectedName);
```

Comment:
It is possible that here there is an error due to location not having the property x.

Reply:
What do you mean by location not having the property X? This code isn't executed on any of the first renders of React. It only gets called when the location dropdown name is changed.

If you're referring to this error, that is present in the handleMouseLeave function and is caused by something else.
![image](https://github.com/TeamLumi/luminescent-team/assets/125856819/946a9cb9-7e87-4ddc-9a71-396ba9f4560f)